File: cpp/001_TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> &numbers, int target) {
        vector<int> ret;
        std::unordered_map<int, int> hash;

        for( size_t i = 0; i < numbers.size(); ++i ){
            std::unordered_map<int, int>::iterator it = hash.find( target - numbers[i] );
            if( it != hash.end() ){
                ret.push_back( hash[target - numbers[i]] );
                ret.push_back( i+1 );
                return ret;
            }
            hash[ numbers[i] ] = i + 1;
        }
        
        return ret;
    }
};

File: js/001_TwoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var ret = [];
    var hash = {};
    for( var i = 0; i < nums.length; ++i ){
        if(hash[target - nums[i]]){
            ret.push(hash[target - nums[i]]);
            ret.push(i + 1);
            return ret;
        } 
        hash[nums[i]] = i + 1;
    }
    return ret;
};